mumd/src/audio.rs

```diff
@@ -216,10 +190,55 @@ impl Audio {
 
         output_stream.play().unwrap();
 
-        let sounds = EVENT_SOUNDS
+        let mut res = Self {
+            output_config,
+            _output_stream: output_stream,
+            _input_stream: input_stream,
+            input_volume_sender,
+            input_channel_receiver: Some(input_receiver),
+            client_streams,
+            sounds: HashMap::new(),
+            output_volume_sender,
+            user_volumes,
+            play_sounds,
+        };
+        res.load_sound_effects(&vec![]);
+        res
+    }
+
+    pub fn load_sound_effects(&mut self, sound_effects: &Vec<SoundEffect>) {
+        let overrides: HashMap<_, _> = sound_effects
             .iter()
-            .map(|(bytes, event)| {
-                let reader = hound::WavReader::new(*bytes).unwrap();
+            .filter_map(|sound_effect| {
+                let (event, file) = (&sound_effect.event, &sound_effect.file);
+                let event = match event.as_str() {
+                    "server_connect" => NotificationEvents::ServerConnect,
+                    "server_disconnect" => NotificationEvents::ServerDisconnect,
+                    "user_connected" => NotificationEvents::UserConnected,
+                    "user_disconnected" => NotificationEvents::UserDisconnected,
+                    "user_joined_channel" => NotificationEvents::UserJoinedChannel,
+                    "user_left_channel" => NotificationEvents::UserLeftChannel,
+                    "mute" => NotificationEvents::Mute,
+                    "unmute" => NotificationEvents::Unmute,
+                    "deafen" => NotificationEvents::Deafen,
+                    "undeafen" => NotificationEvents::Undeafen,
+                    _ => {
+                        warn!("Unknown notification event '{}' in config", event);
+                        return None;
+                    }
+                };
+                Some((event, file))
+            })
+            .collect();
+
+        self.sounds = NotificationEvents::iter()
+            .map(|event| {
+                let bytes_cow = overrides.get(&event)
+                    .map(|file| get_resource(file))
+                    .flatten()
+                    .unwrap_or(Cow::from(include_bytes!("fallback_sfx.wav").as_ref()));
+                let bytes = bytes_cow.as_ref();
+                let reader = hound::WavReader::new(bytes).unwrap();
```

Comment:
Does `bytes` have to be bound to a variable or could it be declared where it's used?